APE10.rst

```diff
@@ -0,0 +1,155 @@
+releases that support only Python 3.
+
+Detailed description
+--------------------
+
+A number of scientific Python projects, including IPython, Matplotlib, Sympy,
+and others have now signed the `Python 3 statement
+<https://python3statement.github.io>`_ which states that scientific Python
+packages will drop support for Python 2 no longer than 2020. In fact, the
+IPython developers have now released `IPython 5.0
+<http://blog.jupyter.org/2016/07/08/ipython-5-0-released/>`_, which will be the
+last major release to support Python 2 - since it is a long term support (LTS)
+release, there will be 5.x bug fix releases until some time in 2019, but future
+major releases that add new features, starting with 6.0 (in 2017), will support
+only Python 3. This APE proposes that the Astropy project should also sign the Python 3 statement
+statement, and provides a target roadmap for this to happen.
+
+Roadmap
+-------
+
+The proposed roadmap for releases, including past releases for reference, is the
+following::
+
+    v1.0 - LTS - February 19, 2015 (supported with bug fixes until June 2017)
+    v1.1 - December 11, 2015
+    v1.2 - June 23, 2016
+    v1.3 - December 2016
+    v2.0 - LTS - June 2017 (supported with bug fixes until end of 2019)
+    v3.0 - December 2017 - first release to support only Python 3+
+    v3.1 - June 2018
+    v3.2 - December 2018
+    v3.3 - June 2019
+    v4.0 - LTS - December 2019 (supported with bug fixes for two years)
+
+Note that v1.0, v2.0, and v4.0 are marked as long-term support releases (LTS),
+which means that they are typically supported with bug fixes for at least two
+years. The above roadmap proposes that the last Python 2-compatible release will
+be v2.0, which we will support with bug fixes until the end of 2019. In the mean
+time, v3.0, to be released in December 2017, will be the first release that does
+not support Python 2.
+
+Note that v3.0 would not be an LTS release, since we would
+otherwise need to maintain two LTS releases in parallel. This is a one-off
+deviation from the version numbering policy outlined in `APE 2
+<https://github.com/astropy/astropy-APEs/blob/master/APE2.rst>`_
+
+Implementation
+--------------
+
+In practice, dropping Python 2.7 support will involve, amongst other things:
+
+* Removing the bundled `six <https://pythonhosted.org/six/>`_ package, and
+  update all the code that used ``astropy.extern.six`` to use the Python 3
+  syntax directly.
+* Removing Python 2.7 from the configuration for the continuous integration
+  services (such as Travis and AppVeyor at the time of writing)
+* Making sure that the documentation on ReadTheDocs is built using Python 3.x
+* Removing any backports that may exist for Python 2.7 compatibility
+* Updating the documentation to remove any mention of Python 2.7, as well as
+  making sure that Python 3.x is not presented as a special case.
+* Remove Python 2 and 2.7 from the PyPI classifiers
+
+Challenges
```

Comment:
Another challenge: maintaining the 2.0 LTS will be more difficult than for 1.0 because development bug fixes will not be guaranteed to be Python-2 compatible.